### ISSUE TITLE
Dust voting threshold issues

### ISSUE THREAD

**Issue by theoreticalbts** [steemit/steem#356]
According to [this user report](https://steemit.com/steemit/@darkminded153/can-not-upvote-on-steemit), 3 SP users aren't able to upvote due to failing [this check](https://github.com/steemit/steem/blob/7c71d291d43d7e8d52c0c7090f081df6f7b10704/libraries/chain/steem_evaluator.cpp#L976-L978).

This is not a problem which requires immediate action, because when HF 14 takes effect on Tuesday, the new voting power equilibrium constants should effectively unblock these accounts.

However, the problem here is that this threshold effectively sets a minimum number of VESTS for an account to be able to vote. Therefore, (some multiple of) this number of VESTS sets a floor on the account registration fee (provided we want new accounts to be able to vote).

Here are some possible solutions:
- (a) Don't solve it. If users don't have enough VESTS and want to be able to vote, then they should buy more VESTS. If the blockchain-level account registration fee, or the registration fee actually paid by steemit.com, falls below the minimum to allow users to be able to vote, so be it -- it's up to the affected users to decide whether they want to buy SP or not.
- (b) Keep the blockchain-level account registration fee, and/or the registration fee actually paid by steemit.com, above the minimum to allow users to be able to vote. If the market price of STEEM increases, this may result in account registration becoming very expensive at some point. Also, we need to set up monitoring of the chain state to timely inform the witnesses and/or steemit.com when they need to raise their reg fees.
- (c) Retroactively correct accounts that were registered with too little SP to vote by `transfer_to_vesting` enough to bring them back above the threshold.
- (d) Allow non-consensus votes, which appear like regular votes in the UI but do not affect payouts and aren't eligible for rewards; the UI should use non-consensus votes whenever the user runs out of voting power and fails this check. (Exactly how the UI should work, how the backend should communicate whether a given account should use consensus or non-consensus voting, and whether/how a user should be able to enable/disable non-consensus voting, are topics for another ticket.) #358
- (e) Set the dust threshold to be a witness-configurable parameter. Requires implementing generic chain parameters. #357

**Comment by abitmore** [comment]
Just a note that HF14 postponed and the change of voting power regeneration has been reverted in 0.14.1, so this issue will remain.

>when HF 14 takes effect on Tuesday, the new voting power equilibrium constants should effectively unblock these accounts.

**Comment by iamsmooth** [comment]
> Keep the blockchain-level account registration fee, and/or the registration fee actually paid by steemit.com, above the minimum to allow users to be able to vote. If the market price of STEEM increases, this may result in account registration becoming very expensive at some point.

This hypothetical seems broken. It should not ever be "very expensive" to reach a minimum voting threshold.

**Comment by mvandeberg** [comment]
Replaced by #1764
